src/coreclr/jit/morph.cpp

```diff
@@ -14090,6 +14090,14 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 op2 = tree->AsOp()->gtOp2;
             }
 
+            // Fold "cmp & 1" to just "cmp"
+            if (tree->OperIs(GT_AND) && tree->TypeIs(TYP_INT) && op1->OperIsCompare() && op2->IsIntegralConst(1))
```

Comment:
sorry, I am late for the review, two questions:
1. do we want to do it when optimizations are disabled?
2. Can op2 or tree be a CSE candidates that we destroy while CSE is keeping links on them?

Reply:
1) I found plenty of such small peepholes in morph without `OptimizationEnabled` guard so it wasn't clear to me when we should really apply it - I guess we want to eventually have three different modes: MinOpts, tier0 (simple opts) and tier1 (all opts).

2) Can op2 or tree be a CSE candidates
op2 is always an integer const (1) and CSE doesn't work for them without a special switch -- so it's not a correct behavior to delete a potential CSE candidate and it might lead to a crash/assert?

Reply:
> op2 is always an integer const (1) and CSE doesn't work for them without a special switch -- so it's not a correct behavior to delete a potential CSE candidate and it might lead to a crash/assert?

the switch is on for arm64 but `tree` can be a CSE candidate without it. I would recommend adding a change to protect against it.